src/parse/parse.js

```diff
@@ -0,0 +1,80 @@
+// Parses bristol fashion text circuits into Circuit objects
+
+'use strict';
+
+/*
+ * Bristol fashion has the following format:
+ * <GATE COUNT> <WIRE_COUNT>
+ * <INPUT COUNT> <FIRST INPUT BIT SIZE> <SECOND INPUT BIT SIZE> ...
+ * <OUTPUT COUNT> <FIRST OUTPUT SIZE> <SECOND OUTPUT SIZE> ...
+ * <GATE INPUT COUNT> <GATE OUTPUT COUNT> <INPUT WIRE INDEX> ... <OUTPUT WIRE INDEX> ... <GATE OPERATION>
+ * ...
+ */
+
+const Circuit = require('../modules/circuit.js');
+const Gate = require('../modules/gate.js');
+
+const RECOGNIZED_OPERATIONS = ['AND', 'XOR', 'INV'];
```

Comment:
We might want to alias `INV` with `NOT` as a valid gate type, because the supposed bristol page says this, and I think logical negation is more commonly called not.

Reply:
cool